### PR TITLE
feat(interactive): Support `Sample()` and `Coin()` steps in GIE

### LINKED ISSUE #3055: feature(Interactive): Add sample step in Gremlin
The Gremlin's [`sample`](https://tinkerpop.apache.org/docs/current/reference/#sample-step)-step, as required by our users, shall be implemented.

### PR DESCRIPTION
**Pull request by BingqingLyu**
## What do these changes do?

As titled. Specifically,

In Compiler:
1. Support `Sample` and `Coin` operator.

In Proto:
1. Add the definition of `Sample`.

In IR-Core:
1. Support to build  `Sample` step in logical & physical plan.

In Runtime:
1. Support `Sample` by number with/without seed.
2. Support `Sample` by ratio with/without seed.
3. unit tests for the new features.

## Related issue number

fixes #3055

### CODE REVIEW

File: docs/interactive_engine/tinkerpop/supported_gremlin_steps.md [hunk 1]
```diff
@@ -503,8 +503,32 @@ Filter the object in the stream given a biased coin toss.
...
+number - allow specified number of objects to pass through the stream.
+```bash
+g.V().sample(10)
+g.V().out().sample(10)
+```
+
+Usages of the with()-step: </br>
+keyValuePair - the options to set repeatable seed.
+```bash
+g.V().sample(10).with('REPEATABLE', 123)
```
Comment (longbinlai) [inline docs/interactive_engine/tinkerpop/supported_gremlin_steps.md:529]: Add info: what is the supported  keys, and what does each key means (e.g. REPEATABLE), what does it mean

File: docs/interactive_engine/tinkerpop/supported_gremlin_steps.md [hunk 2]
```diff
@@ -503,8 +503,32 @@ Filter the object in the stream given a biased coin toss.
...
 ```bash
-g.V().coin(0.2) # range is [0, 1]
+g.V().coin(0.2) # range is [0.0, 1.0]
+g.V().out().coin(0.2)
 ```
+
+Usages of the with()-step: </br>
+keyValuePair - the options to set repeatable seed.
+```bash
+g.V().coin(0.2).with('REPEATABLE', 123)
```
Comment (longbinlai) [inline docs/interactive_engine/tinkerpop/supported_gremlin_steps.md:513]: `REPEATABLE` is not possible in the distributed context. Remove this parameter for now.

Comment (BingqingLyu) [reply docs/interactive_engine/tinkerpop/supported_gremlin_steps.md]: Done.

File: docs/interactive_engine/tinkerpop/supported_gremlin_steps.md (hunk 1 above)
Comment (BingqingLyu) [reply docs/interactive_engine/tinkerpop/supported_gremlin_steps.md]: Removed this.

**Review by longbinlai — APPROVED** [review]

**Review by shirly121 — APPROVED** [review]

**Review by longbinlai — APPROVED** [review]

**Review by longbinlai — DISMISSED** [review]

**Pull request merged**